R/qenv-eval_code.R

```diff
@@ -44,6 +44,18 @@ setMethod("eval_code", signature = c("qenv", "expression"), function(object, cod
         class = c("qenv.error", "try-error", "simpleError"),
         trace = object@code
       )
+    },
+    warning = function(w) {
+      attr(object@code, "warning") <- as.character(w)
```

Comment:
So `object@code` is a vector - we could do `attr(object@code[length(object@code])` to add the warning onto the specific code (and we should do `attr(object@code[length(object@code], "warning") <- c(.., as.character(w)) to catch multiple warnings).

If we're just adding it to the code as a whole then I would actually have object@warnings and store them there instead

Reply:
done